spec/integration/idempotencyKeyChargesSpec.js

```diff
@@ -0,0 +1,53 @@
+const crypto = require('crypto')
+const {Shift4Gateway} = require('../../')
+const cards = require('../data/cards')
+const charges = require('../data/charges')
+const RequestOptions = require('../../lib/requestOptions')
+
+describe('Charges', function () {
+    const api = new Shift4Gateway()
+
+    it('should create one charge without duplicates on multiple requests if idempotency key is used', async () => {
+        // given
+        const requestOptions = new RequestOptions();
+        requestOptions.setIdempotencyKey(crypto.randomUUID())
+        const chargeReq = charges.charge({card: cards.card()})
+
+        // when
+        const createdCharge = await api.charges.create(chargeReq, requestOptions)
+        const sameCharge = await api.charges.create(chargeReq, requestOptions)
+
+        // then
+        expect(sameCharge.id).toEqual(createdCharge.id)
```

Comment:
Can we check here attendance of header: `idempotent-replayed`?

Reply:
We don't pass headers in this SDK to the SDK user, closing for now